File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
prime = [True for _ in range(1000001)]
prime[0]=False
prime[1]=False
j=2
for i in range(2, 1000001):
    if prime[i]:
        j=2
        while j*i<1000001:
            prime[i*j]=False
            j+=1

while True:
    n = int(input())
    if n==0:
        break
    else:
        flag = True
        for i in range(3,n):
            if prime[i] and prime[n-i] == True :
                print("%d = %d + %d"%(n , i , n-i))
                flag = False
                break
        if flag:
            print("Goldbach's conjecture is wrong.")
    

File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n, j int
	var flag bool
	var primes = make([]bool, 1000001)
	for i := 2; i < 1000001; i++ {
		primes[i] = true
	}
	for i := 2; i < 1000001; i++ {
		j = 2
		if primes[i] {
			for i*j < 1000001 {
				primes[i*j] = false
				j++
			}
		}
	}
	for {
		fmt.Fscanf(reader, "%d\n", &n)
		if n == 0 {
			break
		}
		flag = true
		for i := 3; i < n; i++ {
			if primes[i] && primes[n-i] {
				fmt.Fprintf(writer, "%d = %d + %d\n", n, i, n-i)
				flag = false
				break
			}
		}
		if flag {
			fmt.Fprintf(writer, "%s\n", "Goldbach's conjecture is wrong.")
		}
	}
}
